com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -586,5 +586,167 @@ internal NetworkBehaviour GetNetworkBehaviourAtOrderIndex(ushort index)
+            {
+                // Set the position and rotation data marker to true (i.e. flag to know, when reading from the stream, that position and rotation data follows).
+                writer.WriteBool(true);
+
+                // Write position
+                writer.WriteSinglePacked(transform.position.x);
+                writer.WriteSinglePacked(transform.position.y);
+                writer.WriteSinglePacked(transform.position.z);
+
+                // Write rotation
+                writer.WriteSinglePacked(transform.rotation.eulerAngles.x);
+                writer.WriteSinglePacked(transform.rotation.eulerAngles.y);
+                writer.WriteSinglePacked(transform.rotation.eulerAngles.z);
+            }
+            else
+            {
+                // Set the position and rotation data marker to false (i.e. flag to know, when reading from the stream, that position and rotation data *was not included*)
+                writer.WriteBool(false);
+            }
+
+            //Write whether we are including network variable data
+            writer.WriteBool(NetworkManager.NetworkConfig.EnableNetworkVariable);
+
+            if (NetworkManager.NetworkConfig.EnableNetworkVariable)
+            {
+                var buffer = writer.GetStream() as Serialization.NetworkBuffer;
+                // Mark our current position and set the network variable data size initially to zero (NOT as a packed value)
+                var currentStreamPosition = buffer.Position;
+                writer.WriteUInt32(0);
+
+                //Write the network variable data
+                WriteNetworkVariableData(writer.GetStream(), targetClientId);
+
+                //As long as something was written
+                if (buffer.Position > currentStreamPosition)
+                {
+                    //Get the new updated stream buffer position
+                    var updatedPosition = buffer.Position;
+
+                    //Move back to the position just before we wrote our size
+                    buffer.Position = currentStreamPosition;
+
+                    //Write how much data was written
+                    writer.WriteUInt32((uint)(buffer.Position - currentStreamPosition));
+
+                    //Revert the buffer position back to the end of the network variable data written
+                    buffer.Position = updatedPosition;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Ueed to deserialize a serialized scene object which occurs
+        /// when the client is approved or during a scene transition
+        /// </summary>
+        /// <param name="objectStream">inbound stream</param>
+        /// <param name="reader">reader for the stream</param>
+        /// <param name="networkManager">NetworkManager instance</param>
+        static internal void DeserializeSceneObject(Serialization.NetworkBuffer objectStream, PooledNetworkReader reader, NetworkManager networkManager)
```

Comment:
IIRC, the suggested order is `internal static void` instead of `static internal void` (minor)